orttraining/orttraining/python/training/ortmodule/_fallback.py

```diff
@@ -171,7 +172,7 @@ def fallback(self, log_level: _logger.LogLevel, *inputs, **kwargs):
 
         assert self.is_pending(), "`fallback` can only be called when there is a pending fallback"
 
-        if log_level <= _logger.LogLevel.WARNING:
+        if log_level <= _logger.LogLevel.WARNING and not self._raised_ortmodule_onnx_model_exception:
```

Comment:
It would appear that if `retry` is not set, any form of fallback exception will trigger a warning of the fallback.
But we only avoid re-printing in the case of onnx model export exception.

Shall we standardize this behavior for all fallback exceptions? And add a note that the warning will not be raised again if retry is not enabled?

Reply:
I also think we can remove this condition "log_level <= _logger.LogLevel.WARNING", it is not helpful generate duplicate fallback message regardless of log level